src/lib/ota/jethome.ts

```diff
@@ -0,0 +1,93 @@
+const baseurl = 'https://fw.jethome.ru';
+const deviceurl = baseurl + '/api/devices/';
+import * as common from './common';
+import {Logger, Zh, Ota, KeyValueAny} from '../types';
+const axios = common.getAxios();
+import fs from 'fs';
+import path from 'path';
+
+let overrideIndexFileName: string = null;
+let dataDir: string = null;
+
+/**
+ * Helper functions
+ */
+
+async function getFirmwareFile(image: KeyValueAny, logger: Logger) {
+    let urlOrName = image.url;
+
+    // First try to download firmware file with the URL provided
+    if (common.isValidUrl(urlOrName)) {
+        logger.debug(`JetHomeOTA: downloading firmware image from ${urlOrName}`);
+        return await axios.get(urlOrName, {responseType: 'arraybuffer'});
+    }
+
+    logger.debug(`JetHomeOTA: call readLocalFile`);
+    // If the file name is not a full path, then treat it as a relative to the data directory
+    if (!path.isAbsolute(urlOrName) && dataDir) {
```

Comment:
This change can be integrated into `common.ts`, then the function can be removed here.